server.go

```diff
@@ -446,9 +460,15 @@ func (s *WebSocketSession) recvMessages() {
 			return
 		}
 		buf := make([]byte, ioBufferSize)
-		_, err = io.CopyBuffer(ioutil.Discard, r, buf)
-		if err != nil {
-			break
+		if s.server.hasReceiveCallback() {
```

Comment:
`Receiver` is an interface, so we can implement blockholeReceiver.
It removes checking `hasReceiveCallback` every time.

```go
type blockholeReceiver struct {
   pool sync.Pool
}

func newBlockholeReceiver() Receiver {
  return &blockholeReceiver {
    pool: sync.Pool {
        New: func() interface{} { return make([]byte, ioBufferSize) },
    },
  }
}

func (r *receiverCallback) Receive(ctx context.Context, m receivedMessage) error {
  buf := r.pool.Get().([]byte)
  _, err := io.CopyBuffer(ioutil.Discard, m.Message, buf)
  r.pool.Put(buf)
  return err
}
```